aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1641,15 +1641,14 @@ The behavior depends on the dimensionality of the Tensors as follows:
   must be broadcastable).  For example, if tensor1 is a (j x 1 x n x m) Tensor
   and tensor2 is a (k x m x p) Tensor, the returned tensor will be an (j x k x n x p) Tensor.
 */
-Tensor matmul(
-    c10::optional<Tensor> out_opt,
+Tensor _matmul_impl(
+    Tensor& out,
```

Comment:
There actually isn't any reason this can't be a `const Tensor&`, which is better because you never directly overwrite the Tensor.

Reply:
I did this at some point, but it means that I'll need to use lots of const_casts in matmul. Is that preferrable?

Reply:
sort of. The const casts are not good but eventually matmul will get structured-ified so that it doesn't take a mutable reference as argument and then you don't need to const cast. You can leave it like this.